gef-ui/server/controllers/amendments/do-you-have-a-facility-end-date/do-you-have-a-facility-end-date.test.ts

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable import/first */
+const getApplicationMock = jest.fn();
+const getFacilityMock = jest.fn();
+
+import * as dtfsCommon from '@ukef/dtfs2-common';
+import { aPortalSessionUser, CURRENCY, DEAL_STATUS, DEAL_SUBMISSION_TYPE, Facility, PORTAL_LOGIN_STATUS, ROLES } from '@ukef/dtfs2-common';
+import { HttpStatusCode } from 'axios';
+import { createMocks } from 'node-mocks-http';
+import { getDoYouHaveAFacilityEndDate, GetDoYouHaveAFacilityEndDateRequest } from './do-you-have-a-facility-end-date';
+import { Deal } from '../../../types/deal';
+import { DoYouHaveAFacilityEndDateViewModel } from '../../../types/view-models/amendments/do-you-have-a-facility-end-date-view-model';
+
+jest.mock('../../../services/api', () => ({
+  getApplication: getApplicationMock,
+  getFacility: getFacilityMock,
+}));
+
+const dealId = 'dealId';
+const facilityId = 'facilityId';
+const amendmentId = 'amendmentId';
+
+const companyName = 'company name ltd';
+
+const getHttpMocks = () =>
+  createMocks<GetDoYouHaveAFacilityEndDateRequest>({
+    params: { dealId, facilityId, amendmentId },
+    session: {
+      user: { ...aPortalSessionUser(), roles: [ROLES.MAKER] },
+      userToken: 'testToken',
+      loginStatus: PORTAL_LOGIN_STATUS.VALID_2FA,
+    },
+  });
+
+const mockDeal = { exporter: { companyName }, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED } as Deal;
+
+const mockFacility = {
+  currency: {
+    id: CURRENCY.GBP,
+  },
+  hasBeenIssued: true,
+} as Facility;
+
+describe('getDoYouHaveAFacilityEndDate', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.spyOn(dtfsCommon, 'isPortalFacilityAmendmentsFeatureFlagEnabled').mockReturnValue(true);
+
+    getApplicationMock.mockResolvedValue(mockDeal);
+    getFacilityMock.mockResolvedValue({ details: mockFacility });
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
```

Comment:
this line could also go on the top so it is not repeated

Reply:
Whoever picks this up, we should update other amendment tests to do the same, to be consistent.